src/doc/Iterator.ts

```diff
@@ -40,11 +43,15 @@ export default class Iterator {
       if (offset === 0 && length >= nextLine.length) {
         return nextLine;
       } else {
-        return {
+        const id = offset === 0 ? nextLine.id : Line.createId(this.lineIds);
+        const partialLine = {
+          id,
```

Comment:
Adds `id` to `Line` and when splitting a line in two such as when calling `next(10)` then `next()` on a 20-length line, the first call should return the line with its original id and the second call should return a line with a new id.